Extra music and sfx

NathanHinthorne opened the pull request:
Added extra music and sfx files

NathanHinthorne committed:
Merge branch 'main' of https://github.com/NathanHinthorne/good-bad-chad-nathan

NathanHinthorne committed:
Merge branch 'main' into new-br-changes

NathanHinthorne committed:
Merge pull request #9 from NathanHinthorne/new-br-changes

New br changes

NathanHinthorne committed:
Added new SFX

NathanHinthorne committed:
Merge branch 'main' of https://github.com/NathanHinthorne/good-bad-chad-nathan

NathanHinthorne committed:
Prevented the user from selecting other html elements

NathanHinthorne committed:
Made my version of the playground dimension

NathanHinthorne committed:
renamed methods

NathanHinthorne committed:
added SFX and MUSIC to the itil script

NathanHinthorne committed:
swapped out some old sfx. Added new sfx and music

NathanHinthorne committed:
Merge pull request #99 from NathanHinthorne/main

Extra music and sfx

tclaar commented on css/general.css:
```diff
@@ -10,6 +10,9 @@
     text-align: center;
+
+    /* Prevents the user from selecting other html elements. This will fix our issues with un-recorded key presses */
+    user-select: none;
```
Thanks for figuring this out!

tclaar commented on js/classes/entities/midground/weapon/slingshot.js:
```diff
@@ -32,19 +40,34 @@ class Slingshot {
+                ASSET_MGR.playAudio(SFX.SLINGSHOT_LAUNCH4.path, SFX.SLINGSHOT_LAUNCH4.volume);
+                break;
+        }
+
```
I like that you play a random sound effect here! Also, notice that the only difference between each of the cases in your switch is the string used to index `SFX`. You could probably replace the entire switch with something like:
```
const sfx = SFX["SLINGSHOT_LAUNCH" + rand];
ASSET_MGR.playAudio(sfx.path, sfx.volume);
```

tclaar commented on js/scripts/dimension/playground_dimension.js:
Remember that we're not using this script anymore since the Zone refactor. Don't forget to move your music code to that new system.

tclaar commented on js/scripts/main.js:
```diff
@@ -48,7 +48,7 @@ spanishButton.addEventListener("click", () => {
 // (3) Set the current ZONE to be the first one we encounter - village.main.
 
-let ZONE = Zone.getZones().village.main;
+let ZONE = Zone.getZones().playground.nathan;
```
We should be loading village.main, not playground.nathan. I think this should be changed before merging.

tclaar reviewed (changes requested):
Nice work Nathan, thanks for adding sound effect support! Before we merge this pull request, you should have main.js load the `village.main` instead of `playground.nathan`.

NathanHinthorne committed:
Fixed mistake in main

Changed `Zone.getZones().playground.nathan;` to `Zone.getZones().village.main;`

calebkrauter commented on css/general.css:
```diff
@@ -10,6 +10,9 @@
 
 body {
     text-align: center;
+
```
Very nice!

calebkrauter commented on js/classes/asset_manager.js:
```diff
@@ -55,7 +55,7 @@ class AssetManager {
         
                     img.addEventListener("error", () => {
-                        console.log("Error loading " + img.src);
+                        console.log("Error loading " + path);
```
Smart error checking @NathanHinthorne  I like it.

tclaar reviewed (approved):
Thanks for the fix! Nice work.

calebkrauter commented on js/classes/entities/midground/chad.js:
```diff
@@ -179,7 +179,7 @@ class Chad {
         // If Chad can double jump and user is trying to jump than do it!
         if (this.canDoubleJump) {
-            ASSET_MGR.playAudio("./sfx/temp_jump.wav", 0.2);
+            ASSET_MGR.playAudio(SFX.JUMP2.path, SFX.JUMP2.volume);
```
Good idea adding volume here like this. Perhaps we can link all of the sfx volume together and make a volume slider for it and make a separate one for music @NathanHinthorne

calebkrauter commented on js/classes/entities/midground/weapon/slingshot.js:
```diff
@@ -32,19 +40,34 @@ class Slingshot {
+    fire() {
+        ASSET_MGR.stopAudio(SFX.SLINGSHOT_STRETCH.path);
+        
+        // choose from 4 different firing sounds
```
Interesting idea! Nice.

calebkrauter commented on js/classes/entities/midground/weapon/slingshot.js:
```diff
@@ -64,15 +87,17 @@ class Slingshot {
         }, 1000);
+
+        // reset the slingshot stretch sound
+        this.playedStretchSound = false;
```
Ohhhh that's what this sound is very cool.

calebkrauter commented on js/scripts/dimension/playground_dimension.js:
```diff
@@ -73,9 +73,8 @@ const loadPlaygroundDimension = () => {
-        GAME.addEntity(new Portal(new Vector(10, -10), Dimension.LAVA));
-        GAME.addEntity(new Portal(new Vector(10, -15), Dimension.VILLAGE));
-        GAME.addEntity(new Snake(new Vector(-3 * Block.SCALED_SIZE, -20 * Block.SCALED_SIZE)));
+        GAME.addEntity(new Portal(10, -10, Dimension.LAVA));
```
Are we still using Dimension? Might have to update this. @NathanHinthorne

calebkrauter commented on js/scripts/dimension/playground_dimension.js:
```diff
@@ -89,13 +88,24 @@ const loadPlaygroundDimension = () => {
+            ASSET_MGR.playAudio("./music/starting_off_2_sample.wav", 0.1, true);
+        
+            // delete the event listener so that the music doesn't restart when the user clicks again
+            document.body.removeEventListener('click', playMusic);
```
Ah very cool. So is it getting re-added every time you click? @NathanHinthorne

calebkrauter commented on js/scripts/dimension/playground_dimension.js:
```diff
@@ -89,13 +88,24 @@ const loadPlaygroundDimension = () => {
+            document.body.removeEventListener('click', playMusic);
+        };
+        
+        document.body.addEventListener('click', playMusic);
```
Should it play music when you click @NathanHinthorne ? Idk maybe we will want to change it later but not a big deal. Will it trigger every time you shoot?

calebkrauter commented on js/scripts/dimension/playground_dimension.js:
```diff
@@ -89,13 +88,24 @@ const loadPlaygroundDimension = () => {
-        // CHAD.x = -3 * Block.SCALED_SIZE;
-        // CHAD.y = -20 * Block.SCALED_SIZE;
-        CHAD.pos = Vector.blockToWorldSpace(new Vector(-3, -20));
+        CHAD.x = -3 * Block.SCALED_SIZE;
```
Magic numbers, what are -3 and -20?

calebkrauter commented on js/scripts/util.js:
```diff
@@ -41,6 +41,29 @@ const FONT = { // NOTE: the "vt323" part below is not what's assigning the vt323
+ * The game's sound effects.
+ */
+const SFX = {
+    JUMP1: {path: "./sfx/jump1.mp3", volume: 0.2},
```
Should all of these volumes be different? Maybe they should be and this might be great, it would be cool if we can update all of the sfx volumes together after you are done mixing/mastering the individuals sfx sounds to be the right volumes. Nice work! @NathanHinthorne

calebkrauter commented on js/scripts/zone_loaders/playground_zone_loaders.js:
```diff
@@ -39,13 +39,63 @@ const loadPlaygroundEverybody = () => {
+
+        // NOTE: we can't activate music until the user has interacted with the canvas. (this issue is inherent to HTML5)
+        //  If listening for a click is the only way to activate music, that's fine. 
+        //  Our game's START button in the final version can be the trigger.
```
Good idea @NathanHinthorne ! Idea for you Nathan, is there possibly a way to cause an action to start the music as soon as the game loads to simulate a mouse click if that is what is necessary to start the music?

calebkrauter commented on js/scripts/zone_loaders/playground_zone_loaders.js:
```diff
@@ -39,13 +39,63 @@ const loadPlaygroundEverybody = () => {
+            ASSET_MGR.playAudio(MUSIC.PEACEFUL_CHIPTUNE.path, MUSIC.PEACEFUL_CHIPTUNE.volume, true);
+        
+            // delete the event listener so that the music doesn't restart when the user clicks again
+            document.body.removeEventListener('click', playMusic);
```
Is this getting added and removed in the update loop? Idk if we want it too happen to often... @NathanHinthorne

calebkrauter reviewed (approved):
Looks good! Nice work @NathanHinthorne just some minor suggestions for later feature additions and some questions.
